docs/clients/service-providers.md

```diff
@@ -121,3 +121,49 @@ mechanism for instance.
 In order to generate an unsigned transaction (example with
 [coin transfer](https://cosmos.network/rpc/#/ICS20/post_bank_accounts__address__transfers)),
 you need to use the field `generate_only` in the body of `base_req`.
+
+## Cosmos signing
+
+
+Cosmos transaction signing is a fairly simple process. 
+
+Every Cosmos transaction has a canonical json representation. The gaiacli and stargate rest interfaces provides canonical json representations of transactions and their broadcast function will provide compact amino(a protobuf like format) encoding translations.
+
+
+Things to know when signing messages.
+
+The format is 
+```json
+{ 
+"account_number": XXX,
+"chain_id":XXX,
+"fee":XXX,
+"sequence":XXX 
+"memo":XXX,
+"msgs":XXX
+}
+```
+
+The signer must supply chain_id, account number and sequence number.
+
+Fee, msg and memo will be supplied by the transaction composer interface.
+
+`account_number` and `sequence` can be queried directly from the blockchain or cached locally. Getting these numbers wrong is a common cause of invalid signature errors. You can load the mempool of a full node or validator with a sequence of uncommitted transactions with incrementing sequence numbers and it will mostly do the correct thing.  
+
+Before signing, all keys are lexicographically sorted and all white space is removed from the JSON output before signing.
+
+The Signature encoding is the 64-byte concatenation of ECDSArands(ie.r || s), where s is lexicographically less than it’s inverse, to prevent malleability. This is like Ethereum, but without the extra byte for pubkey recovery, since Tendermint assumes the pubkey is always provided anyway.
+
+Singatures and public public key examples
+
+``` json
+[{
+"pub_key":{
+"type":"tendermint/PubKeySecp256k1",
+"value":XXX
+},
+"signature":XXX
+}]
+```
+
+Once signatures are properly generated, insert the json into into the generated transaction and then use the broadcast endpoint.
```

Comment:
```suggestion
Once signatures are properly generated, insert the JSON into into the generated transaction and then use the broadcast endpoint.
```

Reply:
Should we note the difference between asynchronous and synchronous (wait-for-commit) broadcast?

Reply:
Don't know. 

It should probably be documented somewhere